smt.go

```diff
@@ -91,6 +96,54 @@ func (smt *SparseMerkleTree) Get(key []byte) ([]byte, error) {
 	return value, nil
 }
 
+// GetLeaf gets an entire leaf node from the tree.
+func (smt *SparseMerkleTree) GetLeaf(key []byte) (*LeafNode, error) {
+	if bytes.Equal(smt.root, smt.th.placeholder()) {
+		// The tree is empty, return the default value.
+		return nil, nil
```

Comment:
I think returning `nil` for the leaf node should be reserved for the error case. If there's no error, it should be a `&LeafNode{Path: path, ValueHash: nil}`. Also, this behavior should be documented both above the function and the struct declaration.

Reply:
Makes sense :+1: